Common/src/Utils/GraceDelayCancellationTokenSource.cs

```diff
@@ -0,0 +1,83 @@
+// This file is part of the ArmoniK project
+// 
+// Copyright (C) ANEO, 2021-2022. All rights reserved.
+//   W. Kirschenmann   <wkirschenmann@aneo.fr>
+//   J. Gurhem         <jgurhem@aneo.fr>
+//   D. Dubuc          <ddubuc@aneo.fr>
+//   L. Ziane Khodja   <lzianekhodja@aneo.fr>
+//   F. Lemaitre       <flemaitre@aneo.fr>
+//   S. Djebbar        <sdjebbar@aneo.fr>
+//   J. Fonseca        <jfonseca@aneo.fr>
+//   D. Brasseur       <dbrasseur@aneo.fr>
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+// 
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY, without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+// 
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using System.Threading;
+
+namespace ArmoniK.Core.Common.Utils;
+
+public sealed class GraceDelayCancellationTokenSource : IDisposable
+{
+  private readonly TimeSpan                      t1_;
+  private readonly TimeSpan                      t2_;
+  private readonly TimeSpan                      t3_;
+  private readonly TimeSpan                      t4_;
+  private readonly TimeSpan                      t5_;
+  private readonly CancellationTokenRegistration reg_;
+
+  public GraceDelayCancellationTokenSource(CancellationTokenSource source,
+                                           TimeSpan                t1,
```

Comment:
5 timespan?

Reply:
why not ? We made a slightly more general class

Reply:
I would say, if you want more delays, you create more instances of `GraceDelayCancellationTokenSource`.

Reply:
We need one instance with two delays